src/wp-content/themes/twentytwentyone/header.php

```diff
@@ -20,7 +20,17 @@
 	<?php wp_head(); ?>
 </head>
 
-<body <?php body_class(); ?>>
+<body
+	<?php body_class(); ?>
+	data-wp-interactive='{"namespace": "twentytwentyone"}'
+	data-wp-class--primary-navigation-open="state.isPrimaryMenuOpen"
+	data-wp-class--lock-scrolling="state.isPrimaryMenuOpen"
+	data-wp-class--is-dark-theme="state.isDarkMode"
+	data-wp-init--iframes="callbacks.updateWindowWidthOnResize"
+	data-wp-watch--iframes="callbacks.makeIframesResponsive"
+	data-wp-init--darkmode="callbacks.initDarkMode"
+	data-wp-watch--darkmode="callbacks.refreshHtmlElementDarkMode"
```

Comment:
This is fine 🙂

Reply:
My concern is that the `<body>` element isn't in any way "owned" by a specific plugin or theme. Normally, regions can be assumed to be subject to a specific plugin or theme (e.g. a block, or a specific HTML container element), but `<body>` may be manipulated by anyone. For instance, any plugin could add a dynamic body class, in which case it would need to use directives on the body.

How could that work? Wouldn't there be conflicts between namespaces? That's where my concern stems from.

Reply:
I guess it's fine for the theme to set it to its own namespace. After all, the `namespace` option in `data-wp-interactive` is just a shorter way to say "the default namespace of the directives of this element and its children is X (until you find another `data-wp-interactive` that overrides the value)".

Other plugins that want to use directives in the `<body>` element can do so using the non-default syntax to avoid conflicts: `data-wp-class--my-plugin-class="myPlugin::state.someClass"`.

What do you think? Would that be enough?

Reply:
@luisherranz Thanks, I didn't know it was even possible to put namespaces into the attribute values as a prefix. Are both of these methods fully supported? For example, could I choose to not put `data-wp-interactive` anywhere but instead prefix every single action/callback/state reference with the namespace (e.g. `data-wp-class--primary-navigation-open="twentytwentyone::state.isPrimaryMenuOpen"`)?

While generally that would be cumbersome, for something like `<body>` which by definition isn't "owned" by any particular plugin or theme I'm thinking it may be more appropriate to not use a default namespace.

Reply:
`data-wp-interactive` is still required for hydration. It can be empty if you don't want to set a default namespace, though.

By the way, you can use a single string now: `data-wp-interactive="twentytwentyone"` 🙂

Reply:
Beware, it seems that SSR won't work with an empty `data-wp-interactive`.

Reply:
@c4rl0sbr4v0 
> Beware, it seems that SSR won't work with an empty `data-wp-interactive`.

Interesting. Is it specifically that `data-wp-interactive` must not be empty? Or are you saying SSR doesn't work for any directive that doesn't use the same "default" namespace provided by `data-wp-interactive`?

Reply:
`data-wp-interactive data-wp-context="{"a":"b"}"` won't work.
 `data-wp-interactive="" data-wp-context="{"a":"b"}"` neither.

Reply:
That seems to be a quite significant limitation to me. Is there a reasonable way this could be fixed in the Interactivity API?

Reply:
This PR would fix it, but I would love some feedback from @darerodz. As is setting a default 'WP' namespace that every developer could read and use. 

https://github.com/WordPress/wordpress-develop/pull/6262

Reply:
I don't have enough context to help. What implications the namespace have on the client? The fix proposed only changes the default value from `null` to `WP`. Does it mean that devs will have to explicitly use `WP` as a namespace when interacting with stores?

Reply:
No, stores should infer the namespace. But all interactive chunks created with just `data-wp-interactive` would use that default one, being accesible. 

It would create a "rule" and, as @luisherranz mentions, could create unexpected bugs as would have a different behavior than the client.

Let's close that one and work on a better solution.